mapper/orm-jakarta-batch/core/src/main/java/org/hibernate/search/jakarta/batch/core/massindexing/step/beforechunk/impl/BeforeChunkBatchlet.java

```diff
@@ -53,18 +62,41 @@ public String process() throws Exception {
 				MassIndexingJobParameters.PURGE_ALL_ON_START, serializedPurgeAllOnStart,
 				MassIndexingJobParameters.Defaults.PURGE_ALL_ON_START
 		);
+		boolean dropAndCreateSchemaOnStart = SerializationUtil.parseBooleanParameterOptional(
+				MassIndexingJobParameters.DROP_AND_CREATE_SCHEMA_ON_START,
+				serializedDropAndCreateSchemaOnStart,
+				MassIndexingJobParameters.Defaults.DROP_AND_CREATE_SCHEMA_ON_START
+		);
 		boolean mergeSegmentsAfterPurge = SerializationUtil.parseBooleanParameterOptional(
 				MassIndexingJobParameters.MERGE_SEGMENTS_AFTER_PURGE, serializedMergeSegmentsAfterPurge,
 				MassIndexingJobParameters.Defaults.MERGE_SEGMENTS_AFTER_PURGE
 		);
 
-		if ( purgeAllOnStart ) {
+		if ( dropAndCreateSchemaOnStart ) {
+			if ( tenantId != null ) {
+				// If we have a tenant id here, there most likely are other tenants,
+				//   and if we drop-create the schema then we'd lose the indexed docs for other tenants.
+				//   let the user decide what they want to do here, and either remove the tenant filter,
+				//   or do the schema drop through a schema manager.
+				throw log.tenantIdProvidedWithSchemaDrop( tenantId );
+			}
+			JobContextData jobData = (JobContextData) jobContext.getTransientUserData();
+			EntityManagerFactory emf = jobData.getEntityManagerFactory();
+			Search.mapping( emf ).scope( jobData.getEntityTypes() ).schemaManager().dropAndCreate();
+		}
+
+		if ( Boolean.TRUE.equals( dropAndCreateSchemaOnStart )
+				&& ( serializedPurgeAllOnStart != null && Boolean.TRUE.equals( purgeAllOnStart ) ) ) {
+			log.redundantPurgeAfterDrop();
+		}
+
+		// No need to purge if we've dropped-created the schema already
+		if ( purgeAllOnStart && !dropAndCreateSchemaOnStart ) {
 			JobContextData jobData = (JobContextData) jobContext.getTransientUserData();
 			EntityManagerFactory emf = jobData.getEntityManagerFactory();
 			BatchMappingContext mappingContext = (BatchMappingContext) Search.mapping( emf );
-			// TODO : .........
 			PojoScopeWorkspace workspace =
-					mappingContext.scope( Object.class ).pojoWorkspace( Objects.toString( tenantId, null ) );
+					mappingContext.scope( jobData.getEntityTypes() ).pojoWorkspace( Objects.toString( tenantId, null ) );
```

Comment:
I think .... the `Object` scope was targeting a bit too much. It does say purge all, but it probably wasn't meant to target all